rusty-machine/src/learning/gmm.rs

```diff
@@ -132,6 +132,36 @@ impl GaussianMixtureModel {
         }
     }
 
+    /// Constructs a new GMM with the specified prior mixture weights.
+    ///
+    /// The mixture weights must have the same length as the number of components.
+    /// The sum of the mixture weights must be 1.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rusty_machine::learning::gmm::GaussianMixtureModel;
+    /// use rusty_machine::linalg::vector::Vector;
+    ///
+    /// let mix_weights = Vector::new(vec![0.25, 0.25, 0.5]);
+    ///
+    /// let _ = GaussianMixtureModel::with_weights(3, mix_weights);
+    /// ```
+    pub fn with_weights(k: usize, mixture_weights: Vector<f64>) -> GaussianMixtureModel {
+        assert!(mixture_weights.size() == k, "Mixture weights must have length k.");
```

Comment:
I'm not sure the second case you describe would be especially common. Though from this discussion I've realised it is probably best to normalize the mixture weights rather than assert they sum to 1.

I guess I'm struggling to imagine a situation in which the user will discover the result is not `ok` and actually do something about it. If you try to make a new GMM and it fails I'm not sure what you can do? Perhaps you could dump any intermediate data to file before aborting?

With all that said; you probably are right. However, I haven't respected this elsewhere in the library and I would prefer to do this consistently if we bring it in. I'm considering trying it here to see how it goes though.

> But data-mungers who care about ergonomics more than correctness should really be working in Python.

Though I like python - this did make me chuckle :D.